src/daft-sql/src/planner.rs

```diff
@@ -204,15 +204,104 @@ impl SQLPlanner {
+                    }
+                }
+                match (op, set_quantifier) {
+                    // UNION ALL
+                    (SetOperator::Union, SetQuantifier::All) => {
+                        let left = make_query(left);
+                        let right = make_query(right);
+
+                        let left = self.plan_query(&left)?;
+                        let right = self.plan_query(&right)?;
+                        return left.concat(&right).map_err(|e| e.into());
+                    }
+                    (SetOperator::Union, SetQuantifier::Distinct) => {
+                        unsupported_sql_err!("UNION DISTINCT is not supported.")
+                    }
+                    (SetOperator::Union, SetQuantifier::ByName) => {
+                        unsupported_sql_err!("UNION BY NAME is not supported")
+                    }
+                    (SetOperator::Union, SetQuantifier::AllByName) => {
+                        unsupported_sql_err!("UNION ALL BY NAME is not supported")
+                    }
+                    (SetOperator::Union, SetQuantifier::DistinctByName) => {
+                        unsupported_sql_err!("UNION DISTINCT BY NAME is not supported.")
+                    }
+                    (SetOperator::Union, SetQuantifier::None) => {
+                        let left = make_query(left);
+                        let right = make_query(right);
+
+                        let left = self.plan_query(&left)?;
+                        let right = self.plan_query(&right)?;
+                        return left.concat(&right)?.distinct().map_err(|e| e.into());
+                    }
+                    // We use an ANTI join on all columns to implement EXCEPT
+                    (SetOperator::Except, SetQuantifier::None) => {
+                        let left = make_query(left);
+                        let right = make_query(right);
+                        let left = self.plan_query(&left)?;
+                        let right = self.plan_query(&right)?;
+                        let left_schema = left.schema();
+                        let right_schema = right.schema();
+                        if left_schema != right_schema {
+                            invalid_operation_err!("EXCEPT queries must have the same schema")
+                        }
+                        if left_schema.is_empty() {
+                            invalid_operation_err!("EXCEPT queries must have at least one column")
+                        }
+                        let left_on = left_schema
+                            .names()
+                            .into_iter()
+                            .map(|n| col(n.as_ref()))
+                            .collect::<Vec<_>>();
+
+                        let joined = left.join_with_null_safe_equal(
+                            right,
+                            left_on.clone(),
+                            left_on,
+                            None,
```

Comment:
removed the set ops from the PR.

Reply:
@advancedxy I'm not sure we need/want dedicated ops for these. Since all of the set operations _can_ be implemented with joins, I think it makes sense to just have a method on the builder that constructs the proper join. This is similar to how [datafusion does it](https://github.com/apache/datafusion/blob/main/datafusion/expr/src/logical_plan/builder.rs#L1131), and it's what we're doing with `LogicalPlanBuilder::cross_join`. Keeping these as wrappers over the join ops makes optimization a lot easier as well.

So I think having something like `builder.intersect` and `builder.except` is fine, but not `LogicalPlan::{Intersect, Except}`

Reply:
yeah that makes a lot of sense. I don't have any strong opinion on keeping it one way or the other. If theres no objections from anyone else on the team, im happy to go with specialized ops in logicalplan.